app/controllers/concerns/announce_actions/announce_filter.rb

```diff
@@ -19,7 +19,13 @@ def after(controller)
       model = announce_resource(action, controller)
       subject = announce_subject(model)
       message = announce_message(model)
-      announce(subject, announce_action, message)
+      if model.respond_to?(:with_lock)
```

Comment:
In tests at least, this is not always an AR model, so handle when this isn't actually lockable.